qiskit/circuit/library/arithmetic/integer_comparator.py

```diff
@@ -57,9 +57,9 @@ def __init__(
         """
         super().__init__(name=name)
 
-        self._value = None
-        self._geq = None
-        self._num_state_qubits = None
+        self._value: int | None = None
+        self._geq: bool | None = None
+        self._num_state_qubits: int | None = None
```

Comment:
Mainly a question: Do we need the inline typehints here, even though it should be clear from the initializer what the types are? 🙂